src/Microsoft.ML.Transforms/Text/StopWordsRemoverTransform.cs

```diff
@@ -137,97 +125,14 @@ public sealed class Arguments
-                    Contracts.Assert(LangsColIndex >= 0);
-                }
-                else
-                {
-                    _langsColName = null;
-                    LangsColIndex = -1;
-                }
-            }
-
-            /// <summary>
-            /// Binds a text column with the given name using input schema and returns the column index.
-            /// Fails if there is no column with the given name or if the column type is not text.
-            /// </summary>
-            private static void Bind(ISchema input, string name, out int index, bool exceptUser)
-            {
-                Contracts.AssertValue(input);
-                Contracts.AssertValue(name);
-
-                if (!input.TryGetColumnIndex(name, out index))
-                {
-                    throw exceptUser
-                        ? Contracts.ExceptUserArg(nameof(Arguments.Column), "Source column '{0}' not found", name)
-                        : Contracts.ExceptDecode("Source column '{0}' not found", name);
-                }
-
-                var type = input.GetColumnType(index);
-                if (type != TextType.Instance)
-                {
-                    throw exceptUser
-                        ? Contracts.ExceptUserArg(nameof(Arguments.Column), "Source column '{0}' has type '{1}' but must be text", name, type)
-                        : Contracts.ExceptDecode("Source column '{0}' has type '{1}' but must be text", name, type);
-                }
-            }
-
-            public ColInfoEx(ModelLoadContext ctx, ISchema input)
-            {
-                Contracts.AssertValue(ctx);
-                Contracts.AssertValue(input);
-
-                // *** Binary format ***
-                // int: the stopwords list language
-                // int: the id of languages column name
-                Lang = (Language)ctx.Reader.ReadInt32();
-                Contracts.CheckDecode(Enum.IsDefined(typeof(Language), Lang));
-                _langsColName = ctx.LoadStringOrNull();
-                if (_langsColName != null)
-                {
-                    Bind(input, _langsColName, out LangsColIndex, false);
-                    Contracts.Assert(LangsColIndex >= 0);
-                }
-                else
-                    LangsColIndex = -1;
-            }
-
-            public void Save(ModelSaveContext ctx)
-            {
-                // *** Binary format ***
-                // int: the stopwords list language
-                // int: the id of languages column name
-                ctx.Writer.Write((int)Lang);
-                Contracts.Assert((LangsColIndex >= 0 && _langsColName != null)
-                    || (LangsColIndex == -1 && _langsColName == null));
-                ctx.SaveStringOrNull(_langsColName);
-            }
+            public StopWordsRemoverEstimator.Language Language = StopWordsRemoverEstimator.Defaults.DefaultLanguage;
```

Comment:
>StopWordsRemoverEstimator.Language [](start = 19, length = 34)

Actually, when did we come up with this pattern? Where `Columns` are subclasses of `Transformer`, but enums nd defaults are sub-types of `Estimator`. This is sort of weird.

Reply:
We probably need to move ColumnInfo to Estimators but I would rather do it in one bulk, than having it changed in one classes, and not changed in another.

---
In reply to: [227888434](https://github.com/dotnet/machinelearning/pull/1338#discussion_r227888434) [](ancestors = 227888434)